default/python/universe_generation/starsystems.py

```diff
@@ -70,6 +70,16 @@ def name_planets(system):
         fo.set_name(planet, name)
 
 
+def can_have_no_planets(star_type):
+    """
+    Return True is system is allowed to have no planets.
+    This intentionally only checks the STAR_TYPE_MOD_TO_PLANET_SIZE_DIST so that
+    the following code forces most stars to have a planetary body so that in the
+    GUI the presence of a star implies a planet
+    """
+    return universe_tables.STAR_TYPE_MOD_TO_PLANET_SIZE_DIST[star_type][fo.planetSize.noWorld] != 0
```

Comment:
Ah, I see where you're coming from. Well, the values in the `STAR_TYPE_MOD_TO_PLANET_SIZE_DIST` table are modifiers to "dice rolls" for the planet size of each orbit, so the value in the "none" column are not the percent chance of a system being empty. Insofar I wouldn't necessarily expect a greater than zero value there to determine if a star system is allowed to be empty or not, so 0 is more or less as arbitrary a treshold for potentially empty systems as 20.

But that's a minor quibble. My preferred solution would be to decouple the "can be empty" setting for star types from the `STAR_TYPE_MOD_TO_PLANET_SIZE_DIST` table and have an extra table of boolean values for that. Just a vague idea, your solution should work well enough for now.